test/unit/gateways/reach_test.rb

```diff
@@ -0,0 +1,86 @@
+require 'test_helper'
+
+class ReachTest < Test::Unit::TestCase
+  include CommStub
+
+  def setup
+    @gateway = ReachGateway.new(fixtures(:reach))
+    @credit_card = credit_card
+    @amount = 100
+
+    @options = {
+      email: 'johndoe@reach.com',
+      order_id: '123',
+      currency: 'USD',
+      billing_address: {
+        address1: '1670',
+        address2: '1670 NW 82ND AVE',
+        city: 'Miami',
+        state: 'FL',
+        zip: '32191',
+        country: 'US'
+      }
+    }
+  end
+
+  def test_required_merchant_id_and_secret
+    error = assert_raises(ArgumentError) { ReachGateway.new }
+    assert_equal 'Missing required parameter: merchant_id', error.message
+  end
+
+  def test_supported_card_types
+    assert_equal ReachGateway.supported_cardtypes, %i[visa diners_club american_express jcb master discover maestro]
+  end
+
+  def test_should_be_able_format_a_request
+    post = {
+      request: { someId: 'abc123' },
+      card: { number: '12132323', name: 'John doe' }
+    }
+
+    formatted = @gateway.send :format_and_sign, post
+
+    refute_empty formatted[:signature]
+    assert_kind_of String, formatted[:request]
+    assert_kind_of String, formatted[:card]
+
+    assert_equal 'abc123', JSON.parse(formatted[:request])['someId']
+    assert_equal '12132323', JSON.parse(formatted[:card])['number']
+    assert formatted[:signature].present?
+  end
+
+  def test_successfully_build_a_purchase
+    stub_comms do
+      @gateway.authorize(@amount, @credit_card, @options)
+    end.check_request do |_endpoint, data, _headers|
+      request = JSON.parse(URI.decode_www_form(data)[0][1])
+      card = JSON.parse(URI.decode_www_form(data)[1][1])
+
+      # request
+      assert_equal request['ReferenceId'], @options[:order_id]
+      assert_equal request['Consumer']['Email'], @options[:email]
+      assert_equal request['ConsumerCurrency'], @options[:currency]
+      assert_equal request['Capture'], false
+
+      # card
+      assert_equal card['Number'], @credit_card.number
+      assert_equal card['Name'], @credit_card.name
+      assert_equal card['VerificationCode'], @credit_card.verification_value
+    end.respond_with(successful_purchase_response)
+  end
+
+  def test_properly_set_capture_flag_on_purchase
+    stub_comms do
+      @gateway.purchase(@amount, @credit_card, @options)
+    end.check_request do |_endpoint, data, _headers|
+      request = JSON.parse(URI.decode_www_form(data)[0][1])
+      assert_equal true, request['Capture']
+    end.respond_with(successful_purchase_response)
+  end
+
+  private
+
+  def successful_purchase_response
```

Comment:
Mmm I tried @naashton  something like that,  but I  got decode errors because `stub_com` is sutubbing the call to `ssl_post` and the result is going to be passed to the `parse` method, so if I set the stubbed return value to something that is already parsed I'm going to get decode error.

The only way that I can think of to implement this is to stub out the `parse` method but i'm not sure if formatting is worth no using the real code that handles the response.

maybe you know  other option/angle to implement that ?